packages/api/src/controllers/room.ts

```diff
@@ -110,10 +113,18 @@ app.post(
   authorizer({}),
   validatePost("room-egress-payload"),
   async (req, res) => {
-    const room = await getRoom(req);
+    await getRoom(req);
 
-    if (room.egressId !== undefined && room.egressId != "") {
```

Comment:
Another approach here could just be to fetch the egress by ID and check if `isEgressRunning` as well, before assuming it is. I like having the reference to the external objects in our DB, also keeping it consistent with the rooms referencing behavior themselves, and it might be more efficient after many recorded calls are made on the same room. 

Just a suggestion tho, not a must. Any perf issues will take a couple years to happen on any room lol and the rest is non-behavioral.

Reply:
Thanks for questioning this, since livekit said to me that egress was a per session thing I didn't think they would build up historically in these API responses but turns out they do. The livekit API has some slightly odd quirks, I've just found that despite the egress info being preserved, if you try to start an egress for a room without an active session it just tells you that the room doesn't exist! That causes another bug for us so I will fix that in this PR!

Reply:
Livekit commented that the API will only return up to 24 hours worth of results so no worries about an infinitely growing list over time 👍